autosklearn/data/feature_validator.py

```diff
@@ -294,10 +294,10 @@ def _check_data(
 
         # Then for Pandas, we do not support Nan in categorical columns
         if hasattr(X, "iloc"):
-            enc_columns, _ = self._get_columns_to_encode(X)
-            if len(enc_columns) > 0:
+            self.enc_columns, _ = self._get_columns_to_encode(X)
```

Comment:
Hey, could you please add an explanation why this changed?

Reply:
Yes, I added an explanation in code but let me elaborate here.

The enc_columns can only be set during fit(). So it is not possible to change this unless you refit the whole data, which then makes this checking unnecessary.